src/BreakpointViewer.cpp

```diff
@@ -860,18 +850,14 @@ std::optional<Breakpoint> BreakpointViewer::parseTableRow(BreakpointRef::Type ty
 void BreakpointViewer::onSymbolTableChanged()
 {
 	for (int row = 0; row < bpTableWidget->rowCount(); ++row) {
-		auto* item = bpTableWidget->item(row, BP_ADDRESS);
+		auto* item = bpTableWidget->item(row, LOCATION);
 
-		// scan tooltip validity
-		if (!item->text().isEmpty()) {
-			Symbol* s = debugSession.symbolTable().getAddressSymbol(item->text());
-			auto address = stringToValue<uint16_t>(item->text());
-			assert(address);
+		if (auto address = stringToValue<int>(item->text())) {
+			Symbol* s = debugSession.symbolTable().getAddressSymbol(*address);
 
-			if (!s || *address != s->value()) {
-				setTextField(BreakpointRef::BREAKPOINT, row, LOCATION, item->text());
-				setTextField(BreakpointRef::BREAKPOINT, row, BP_ADDRESS, "");
-			}
+			setTextField(BreakpointRef::BREAKPOINT, row, LOCATION,
+			             s ? s->text() : item->text(),
+			             s ? item->text() : "");
 		}
 	}
 }
```

Comment:
I don't think the `onSymbolTableChanged()` function is correct in case:
* A label is removed from the symbol table (and that label was used in a breakpoint). In this case the breakpoint viewer should revert to a different label with the same value, or the numeric value itself.
* The numeric value of a label was changed. Should be treated as above (as if the label was removed). Or do we want to keep the label, but move the breakpoint to the new address? (My preference would be to not move the label).

This implementation might also change one label into another label with the same numeric value. Maybe this is acceptable, but if it's easy to avoid then try to do that.